enterprise/api_client/sso_orchestrator.py

```diff
@@ -0,0 +1,140 @@
+        self.session = None
+
+    def _get_orchestrator_callback_url(self, config_pk):
+        """
+        get the callback url for the SSO Orchestrator API
+        """
+        lms_base_url = get_configuration_value('LMS_ROOT_URL', settings.LMS_ROOT_URL)
+        if not lms_base_url:
+            raise ClientError(
+                "Failed to get LMS_BASE_URL from configuration.",
+                HTTPStatus.INTERNAL_SERVER_ERROR.value
+            )
+        path = reverse(
+            'enterprise-customer-sso-configuration-orchestration-complete',
+            kwargs={'configuration_uuid': config_pk},
+        )
+        return f"{lms_base_url}{path}"
+
+    def _get_orchestrator_configure_url(self):
+        """
+        get the configure url for the SSO Orchestrator API
+        """
+        # probably want config value validated for this
+        return urljoin(self.base_url, get_sso_orchestrator_configure_path()) 
+
+    def _create_auth_header(self):
+        """
+        create the basic auth header for requests to the SSO Orchestrator API 
+        """
+        orchestrator_username = get_sso_orchestrator_basic_auth_username()
+        if not orchestrator_username:
+            raise ClientError(
+                "Failed to create SSO Orchestrator auth headers: username required.",
+                HTTPStatus.INTERNAL_SERVER_ERROR.value
+            )
+        orchestrator_password = get_sso_orchestrator_basic_auth_password()
+        if not orchestrator_password:
+            raise ClientError(
+                "Failed to create SSO Orchestrator auth headers: password required.",
+                HTTPStatus.INTERNAL_SERVER_ERROR.value
+            )
+        return f"Basic {base64.b64encode(f'{orchestrator_username}:{orchestrator_password}'.encode('utf-8')).decode()}"
+
+    def _create_session(self):
+        """
+        create a requests session object
+        """
+        if not self.session:
+            self.session = requests.Session()
+            self.session.headers.update({
+                "Authorization": self._create_auth_header(),
+            })
+
+    # No real need for this method at the moment as the orchestrator only supports post
+    # to /configure
+    # def _get(self, url, params=None):
+    #     """
+    #     make a GET request to the SSO Orchestrator API
+    #     """
+    #     self._create_session()
+    #     response = self.session.get(url, params=params)
+    #     if response.status_code != HTTPStatus.OK.value:
+    #         raise ClientError(
+    #             f"Failed to get data from {url}: {response.text}",
+    #             response.status_code
+    #         )
+    #     return response.json()
+
+    def _post(self, url, data=None):
+        """
+        make a GET request to the SSO Orchestrator API
+        """
+        self._create_session()
+        response = self.session.post(url, data=data)
+        if response.status_code != HTTPStatus.OK.value:
```

Comment:
nit: you may want to have a wider range of 'success' such as `status_code < 400`

Reply:
I think it's the other way 'round and we're filtering for failed requests so shouldn't it be `status_code >= 300`?